src/modules/m_nationalutf.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2014 Googolplexed <googol@googolplexed.net>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "inspircd.h"
+#include "caller.h"
+
+class ModuleNationalUTF : public Module,public HandlerBase1<bool, const std::string&>
+{
+	caller1<bool, const std::string&> rememberer;
+	struct Range {
+		unsigned long min;
+		unsigned long max;
+		Range(unsigned long mi,unsigned long ma)
+		{
+			min = mi;
+			max = ma;
+		};
+	};
+	std::vector<Range> ranges;
+
+	//Returns the next character in UTF32 encoding and advances the iterator
+	//past it. Returns 0 on failure
+	unsigned long utf8to32(std::string str,std::string::const_iterator& it)
```

Comment:
Either pass `str` by const ref or pass the .end() iterator if that's all what you use `str` for